scripts/deploy.sh

```diff
@@ -50,14 +50,23 @@ if [[ "${DEPLOY_AIO}" == "yes" ]]; then
   fi
   # force the deployment of haproxy for an AIO
   export DEPLOY_HAPROXY="yes"
-  if [[ ! -d /etc/openstack_deploy/ ]]; then
+  if [[ ! -e /openstack ]]; then
     ./scripts/bootstrap-aio.sh
     pushd ${RPCD_DIR}
-      for filename in $(find etc/openstack_deploy/ -type f -iname '*.yml'); do
-        if [[ ! -a "/${filename}" ]]; then
-          cp "${filename}" "/${filename}";
+      for relative_path in $(find etc/openstack_deploy/ -type f -iname '*.yml'); do
+        # If relative_path exists in /etc/openstack_deploy then it is merged
+        # into then repo version overriding the repo value when there is a
+        # conflict.
+        if [[ -e "/${relative_path}" ]]; then
+          # Output is written to the base file
+          #                         Base                 Override
+          ../scripts/update-yaml.py "${relative_path}"   "/${relative_path}";
```

Comment:
This isn't about OSA vs RPC. At this point OSA hasn't been checked out yet. What this means is that if you create /etc/openstack_deploy and put some configs in there before running deploy.sh, those values will take precedence over values that are in the rpc-openstack repo. This allows overriding values in the gate. 

What actually happens is that the repo version is read, its updated with any values from /etc, then the repo version is written out to disk. That means the merged version is now in tree, so its copied out to etc, then reset via git so we don't leave a dirty repo.